src/adminAuth.module.ts

```diff
@@ -1,18 +1,42 @@
-import { TypeOrmModule } from '@nestjs/typeorm'
+import { TypeOrmModule, getRepositoryToken } from '@nestjs/typeorm'
 import { Module } from '@nestjs/common'
 import AdminUserEntity from './adminUser.entity'
 import { AdminUserService } from './adminUser.service'
 import { LocalStrategy } from './local.strategy'
 import { AdminUserController } from './adminUser.controller'
 import { AdminCoreModuleFactory } from './adminCore.module'
 import { injectionTokens } from './tokens'
-import { AdminAuthenticatorConstructor } from './adminAuthenticator.interface'
+import AdminUser from './adminUser.entity'
+import { Repository } from 'typeorm'
 
 const defaultCoreModule = AdminCoreModuleFactory.createAdminCoreModule({})
 
-export interface AdminAuthModuleConfig {
-  adminCoreModule?: any
-  adminAuthInterface?: AdminAuthenticatorConstructor
+export type Authenticator = (email: string, pass: string) => object | null | Promise<object | null>
+
+export interface AdminAuthenticator {
+  imports: any[]
+  useFactory: (dep: any) => Authenticator
+  inject: any[]
+}
+
+export const AdminUserAuthenticator = {
+  imports: [TypeOrmModule.forFeature([AdminUser])],
+  // @ts-ignore
+  useFactory: (adminUserRepository: Repository<AdminUser>) => {
+    return async function validateCredentials(email: string, pass: string) {
+      const user: AdminUser | null = await adminUserRepository.findOne(email)
```

Comment:
I dont think this works (the password is hashed). You'd want to inject the `AdminUserService` and call its `validate` method